test/e2e/localvolumediscovery_test.go

```diff
@@ -225,7 +225,8 @@ func verifyLocalVolumeDiscoveryResult(t *testing.T, lvdr *localv1alpha1.LocalVol
 		}
 
 		if len(lvdr.Status.DiscoveredDevices) == 0 {
-			return false, fmt.Errorf("failed to get the discovered devices")
+			t.Errorf("waiting for discovered devices to be populated")
+			return false, nil
```

Comment:
Fixed an e2e flake I hit twice with this. My e2e passes without this, but the CI one hit it twice.

@sp98 I fixed what I think is a flake with lvdiscoveryresult e2e, so can you review this part? thanks :)

I'm not sure what discoveredDevices are expected here because `verifyLocalVolumeDiscoveryResult` is first called before `createAndAttachAWSVolumes`. Maybe it's the OS disks being listed but marked unavailable?

Reply:
It passed locally, but the [CI e2e](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_local-storage-operator/208/pull-ci-openshift-local-storage-operator-master-e2e-operator/1361989118099197952) hit:
```
     --- FAIL: TestLocalStorage/LocalVolumeDiscovery (36.36s)
        localvolume_test.go:358: Node ip-10-0-150-66.us-west-2.compute.internal updated successfully
        localvolume_test.go:323: Waiting for availability of diskmaker-discovery daemonset
        localvolume_test.go:323: Waiting for availability of diskmaker-discovery daemonset
        localvolume_test.go:331: Waiting for full availability of diskmaker-discovery daemonset (0/1)
        localvolume_test.go:331: Waiting for full availability of diskmaker-discovery daemonset (0/1)
        localvolume_test.go:337: Daemonset available (1/1)
        localvolumediscovery_test.go:84: error verifying localvolumediscoveryresult. failed to get the discovered devices
        localvolume_test.go:358: Node ip-10-0-150-66.us-west-2.compute.internal updated successfully
    main_test.go:126: running 1 cleanup functions. 
```